File: mobile/lib/models/question.dart
class Question {
  final int id;

  // Existing fields
  final String asker;
  final String target;
  final String text;
  final bool hasAnswer;
  final String? answer;

  // 🔥 NEW fields (Feature E)
  final int? roundLobbyId;
  final String? targetName;
  final String? askerName;

  Question({
    required this.id,
    required this.asker,
    required this.target,
    required this.text,
    required this.hasAnswer,
    required this.answer,

    // NEW
    this.roundLobbyId,
    this.targetName,
    this.askerName,
  });

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['id'],

      // existing mappings
      asker: json['asker'] ?? json['asker_name'] ?? "",
      target: json['target'] ?? json['target_name'] ?? "",
      text: json['text'],
      hasAnswer: json['has_answer'],
      answer: json['answer'],

      // NEW mappings from backend
      roundLobbyId: json['round_lobby_id'],
      targetName: json['target_name'],
      askerName: json['asker_name'],
    );
  }
}

File: backend/app/Models/Question.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Question extends Model
{
    protected $fillable = [
        'round_id',
        'asker_id',
        'target_id',
        'text',
        'answer',
        'answered_at'
    ];

    protected $appends = [
        'round_lobby_id',
        'target_name',
        'asker_name',
    ];

    public function round()
    {
        return $this->belongsTo(Round::class);
    }

    public function asker()
    {
        return $this->belongsTo(Player::class, 'asker_id');
    }

    public function target()
    {
        return $this->belongsTo(Player::class, 'target_id');
    }

    // ---------------------------------------------
    // FEATURE E – Needed by Flutter
    // ---------------------------------------------

    public function getRoundLobbyIdAttribute()
    {
        // round belongs to lobby → round->lobby_id
        return $this->round ? $this->round->lobby_id : null;
    }

    public function getTargetNameAttribute()
    {
        return $this->target ? $this->target->name : null;
    }

    public function getAskerNameAttribute()
    {
        return $this->asker ? $this->asker->name : null;
    }
}
